torch/csrc/jit/passes/onnx/peephole.cpp

```diff
@@ -623,6 +623,45 @@ static void eraseListConstruct(Block* block, int opset_version) {
   eraseListConstruct(block->return_node(), opset_version);
 }
 
+static void eraseListUnpack(Block* block, int opset_version);
+
+static void eraseListUnpack(Node* n, int opset_version) {
+  for (auto b : n->blocks()) {
+    eraseListUnpack(b, opset_version);
+  }
+
+  if (n->kind() == prim::ListUnpack) {
+    auto g = n->owningGraph();
+    for (size_t i = 0; i < n->outputs().size(); ++i) {
+      Node* seq_idx_n = g->create(onnx::Constant, 1);
+      seq_idx_n->t_(attr::value, at::scalar_to_tensor(at::Scalar(int64_t(i))));
+      seq_idx_n->insertBefore(n);
+
+      Node* seq_at_n = g->create(onnx::SequenceAt, 1);
+      seq_at_n->addInput(n->input());
+      seq_at_n->addInput(seq_idx_n->output());
+      seq_at_n->output()->setType(n->output(i)->type());
+      seq_at_n->insertBefore(n);
+      n->output(i)->replaceAllUsesWith(seq_at_n->output());
+    }
+  }
+}
+
+static void eraseListUnpack(Block* block, int opset_version) {
+  if (opset_version < OPSET_VERSION_11) {
+    // onnx::SequenceAt was introduced in onnx opset version 11
```

Comment:
If the model has `prim::ListUnpack`, and for opset < 11, seems that we cannot export.
Then can we throw a warning message to say it is not supported until opset 11+?

Reply:
Added error message. Also added DCE before this pass, so that unused `prim::ListUnpack` are ensured to be removed beforehand to not trigger false error.